src/features/instance/applications/editor/components/TextEditorView/index.tsx

```diff
@@ -28,9 +31,45 @@ export function TextEditorView() {
 	const { selectedFolderFile, onSaveFile, isSavingFile, isFolder } = useEditorView();
 	const [language, setLanguage] = useState('javascript');
 	const [updateFileContent, setUpdateFileContent] = useState<string>(selectedFolderFile.content || '');
+	const { clusterId, instanceId } = useParams({ strict: false });
+	const targetId = instanceId ?? clusterId;
+	const targetNoun = instanceId ? 'Instance' : 'Cluster';
+	const { mutate: restartInstance, isPending: isRestartInstancePending } = useUpdateRestartInstance();
 
 	const crumbPath = selectedFolderFile.filePath.split('/').slice(1).join('/').replace(/\//g, ' > ');
 
+	const restartingInstance = () => {
```

Comment:
I suppose `restartInstance` is a bit of a misnomer since it is being used to restart a cluster and _not_ an instance when in cluster mode, right?